numba/cuda/cudaimpl.py

```diff
@@ -295,10 +301,94 @@ def ptx_atomic_add_tuple(context, builder, sig, args):
     if aryty.dtype == types.float32:
         lmod = cgutils.get_module(builder)
         return builder.call(nvvmutils.declare_atomic_add_float32(lmod), (ptr, val))
+    elif aryty.dtype == types.float64:
+        lmod = cgutils.get_module(builder)
+        return builder.call(nvvmutils.declare_atomic_add_float64(lmod), (ptr, val))
     else:
         return builder.atomic_rmw('add', ptr, val, 'monotonic')
 
 
+@register
+@implement(stubs.atomic.max, types.Kind(types.Array), types.intp, types.Any)
+def ptx_atomic_add_intp(context, builder, sig, args):
+    aryty, indty, valty = sig.args
+    ary, ind, val = args
+    dtype = aryty.dtype
+
+    if dtype != valty:
+        raise TypeError("expect %s but got %s" % (dtype, valty))
+    if aryty.ndim != 1:
+        raise TypeError("indexing %d-D array with 1-D index" % (aryty.ndim,))
+
+    lary = context.make_array(aryty)(context, builder, ary)
+    ptr = cgutils.get_item_pointer(builder, aryty, lary, [ind])
+
+    if dtype == types.float64:
+        lmod = cgutils.get_module(builder)
+        return builder.call(nvvmutils.declare_atomic_max_float64(lmod), (ptr, val))
+    else:
+        raise TypeError('Unimplemented atomic max with %s array' % dtype)
+
+
+@register
+@implement(stubs.atomic.add, types.Kind(types.Array),
```

Comment:
If removing `ptx_atomic_max_unituple` allows the tests to carry on passing, should I just leave it out and let the `UniTuple` case be handled by the `Tuple` case?